### PR TITLE
Add SourceRootMappedPathsFeatureSupported property

### PR DESCRIPTION
**Pull request by tmat**
### Customer scenario

Add a property to `Microsoft.Managed.Core.targets` that declares that target `InitializeSourceRootMappedPaths` is available. This property allows SourceLink package targets to detect whether or not to depend on this target and avoid the dependency if the compiler doesn't support the target (e.g. F# projects, projects using older C# compiler version). Without this property available SourceLink package would fail with with a cryptic build error that doesn't tell the customer anything useful.

### Bugs this fixes

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/598723

### Workarounds, if any

No reasonable workaround.

### Risk

Small.

### Performance impact

None.

### How was the bug found?

SourceLink package testing.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review.

**Comment by tmat** [comment]
I'll update the targets tests shortly to validate the property is set ...

**Comment by tmat** [comment]
@nguerrera PTAL

File: src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets [hunk 1]
```diff
@@ -88,7 +88,14 @@
       <SourceRoot Include="@(_MappedSourceRoot)" />
     </ItemGroup>
   </Target>
-  
+
+  <!-- 
+    Declare that target InitializeSourceRootMappedPaths that populates MappedPaths metadata on SourceRoot items is available.
+  -->
+  <PropertyGroup>
+    <SourceRootMappedPathsFeatureSupported>true</SourceRootMappedPathsFeatureSupported>
```
Comment (agocke) [inline src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: Do you want to condition on this property not already being set?

Comment (tmat) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: Why? This should be true if the target is present. It isn't meant to be overridable by the project.

Comment (tmat) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: Ideally msbuild would have `exists` function that one can pass target name to. This is just a workaround for lack of such feature.

Comment (tmat) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: https://github.com/Microsoft/msbuild/issues/3090

Comment (tmat) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: See also https://github.com/Microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L748 that uses the same pattern.

**Comment by tmat** [comment]
Approved by Barry. Merging.

**Pull request merged**